app/controllers/api/v3/bikes.rb

```diff
@@ -0,0 +1,176 @@
+          find_bike
+        end
+
+
+        desc "Add a bike to the Index!<span class='accstr'>*</span>", {
+          authorizations: { oauth2: [{ scope: :write_bikes }] },
+          notes: <<-NOTE
+            **Requires** `write_bikes` **in the access token** you use to create the bike.
+
+            <hr> 
+
+            **Creating test bikes**: To create test bikes, set `test` to true. These bikes:
+
+            - Do not show turn up in searches
+            - Do not send an email to the owner on creation
+            - Are automatically deleted after a few days
+            - Can be viewed in the API /v2/bikes/{id} (same as non-test bikes)
+            - Can be viewed on the HTML site /bikes/{id} (same as non-test bikes)
+
+            *`test` is automatically marked true on this documentation page. Set it to false it if you want to create actual bikes*
+
+            **Ownership**: Bikes you create will be created by the user token you authenticate with, but they will be sent to the email address you specify.
+
+          NOTE
+        }
+        params do
+          requires :serial, type: String, desc: "The serial number for the bike"
+          requires :manufacturer, type: String, desc: "Manufacturer name or ID"
+          # [Manufacturer name or ID](api_v2#!/manufacturers/GET_version_manufacturers_format)
+          requires :owner_email, type: String, desc: "Owner email"
+          requires :color, type: String, desc: "Main color or paint - does not have to be one of the accepted colors"
+          optional :test, type: Boolean, desc: "Is this a test bike?"
+          optional :organization_slug, type: String, desc: "Organization bike should be created by. **Only works** if user is a member of the organization"
+          optional :cycle_type_name, type: String, values: CYCLE_TYPE_NAMES, default: 'bike', desc: "Type of cycle (case sensitive match)"
+          use :bike_attrs
+          optional :components, type: Array do
+            use :components_attrs
+          end
+        end
+        post '/', serializer: BikeV2ShowSerializer, root: 'bike' do
+          declared_p = { "declared_params" => declared(params, include_missing: false).merge(creation_state_params) }
+          b_param = BParam.create(creator_id: creation_user_id, params: {'bike': declared_p['declared_params']}, origin: 'api_v3')
+          ensure_required_stolen_attrs(b_param.params)
+          # Prevents unnecessary DB hits
+          if b_param.errors.present? or b_param.bike_errors.present?
+            e = b_param.errors
+            return error!(e.full_messages.to_sentence, 401)
+          end
+
+          existing_bike = Bike.find_by(serial_normalized: SerialNormalizer.new({serial: b_param[:serial]}).normalized)
+
+          # Assume only one bike for now, but may need to check len of existing bike later depending on what Seth says
+          if existing_bike.present?
+            # Does this check secondary emails?
+            authorize_bike_for_user
```

Comment:
Does this function take care of looking at all secondary emails of your owners? I'm also concerned about the secondary DB access here -- I was trying to figure out a way to combine it all into one on line 150, but I couldn't figure out a good way to do so since it looks like you would have to join from Bikes through Ownerships to Users.